### ISSUE TITLE
Metrics should not depend on MRE

### ISSUE THREAD

**Issue by eakuefner** [catapult-project/catapult#2096]
Tiny design manifesto: https://docs.google.com/document/d/1aAB-d52E2y28BqADqokKhq29KhXk_IhAiXS86fztOko/edit?usp=sharing

TL;DR: we should move `metric_runner` and its dependency (`metric_map_function`) out of `tracing/metrics/` and into `common/metric_runner` for two reasons:

1. It's a circular dependency right now (MRE on tracing, tracing on MRE through metrics)
2. `tracing/metrics/` should be conceptually separate from `metric_runner`, whose major purpose is to integrate with MRE.

I think we probably want to block this on #1962 because we shouldn't stand up our thousand-line support package just to independently support these two files.

Fortunately, this is not a circular dependency in practice, at least not yet. Let's try to resolve this before it shoots somebody in the foot :)

@anniesullie @nedn @benshayden @natduca

**Comment by natduca** [comment]
Goal sounds good. Godspeed :)

**Comment by eakuefner** [comment]
I think we've realized that this is a non-goal. Instead, the current plan is to eventually migrate MRE into tracing/ as-is.
